django_mongodb/compiler.py

```diff
@@ -473,12 +451,47 @@ def get_project_fields(self, columns=None, ordering=None):
                 if self.query.alias_refcount[alias] and self.collection_name != alias:
                     fields[alias] = 1
             # Add order_by() fields.
-            for column, _ in ordering or []:
-                foreign_table = column.split(".", 1)[0] if "." in column else None
-                if column not in fields and foreign_table not in fields:
-                    fields[column] = 1
+            for alias, expression in ordering or []:
+                nested_entity = alias.split(".", 1)[0] if "." in alias else None
+                if alias not in fields and nested_entity not in fields:
+                    fields[alias] = expression.as_mql(self, self.connection)
         return fields
 
+    def _get_ordering(self):
+        """
+        Process the query's OrderBy objects and return:
+        - A tuple of ('field_name': Col/Expression, ...)
+        - A bson.SON mapping to pass to $sort.
+        - A tuple of ('field_name': Expression, ...) for expressions that need
+          to be added to extra_fields.
+        """
+        fields = {}
+        sort_ordering = SON()
+        extra_fields = {}
+        idx = itertools.count(start=1)
+        for order in self.order_by_expressions or []:
+            if isinstance(order.expression, Ref):
+                # TODO change?
```

Comment:
todo: understand this to know whether it makes a difference if we use the commented version or the uncommented version.

Reply:
There are no differences.  is okey to just use refs, but maybe we broke an abstraction? but in all part of the code we do (if this is the case)

Reply:
🤔 mmh now I see the idea, and you are right, we could ignore the refs projection, we don't need them, because they are already managed in $project. So we can remove the refs from the extra fields